synnax/pkg/distribution/framer/relay/streamer.go

```diff
@@ -31,13 +31,13 @@ type streamer struct {
 
 type StreamerConfig = Request
 
-func (r *Relay) NewStreamer(_ context.Context, cfg StreamerConfig) (Streamer, error) {
-	return &streamer{
-		keys:    lo.Uniq(cfg.Keys),
-		addr:    address.Rand(),
-		demands: r.demands,
-		relay:   r,
-	}, nil
+func (r *Relay) NewStreamer(ctx context.Context, cfg StreamerConfig) (Streamer, error) {
+	keys := lo.Uniq(cfg.Keys)
+	// Check that all keys exist.
+	if err := r.cfg.ChannelReader.NewRetrieve().WhereKeys(keys...).Exec(ctx, nil); err != nil {
```

Comment:
so this just checks that all the channels  exist before you open a streamer? is there a test in go for this?

Reply:
The client libraries test this code but I will add a test in the server